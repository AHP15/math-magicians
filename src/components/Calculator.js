import React from 'react';
import PropTypes from 'prop-types';

import CalculatorButton from './CalculatorButton';
import Result from './Result';

import '../styles/Calculator.css';

// eslint-disable-next-line react/prefer-stateless-function
export default class Calculator extends React.Component {
  // eslint-disable-next-line no-useless-constructor
  constructor(props) {
    super(props);
  }

  render() {
    const { buttons } = this.props;
    return (
      <div className="calculator">
        <Result result={0} />
        {
          buttons.map(({ text, changeBg }) => (
            <CalculatorButton key={text} text={text} changeBg={changeBg} />
          ))
        }
      </div>
    );
  }
}

Calculator.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  buttons: PropTypes.array.isRequired,
};
