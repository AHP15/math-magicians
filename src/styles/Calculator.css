.calculator {
  width: fit-content;
  height: 500px;
  display: grid;
  grid-template-columns: repeat(4, 60px);
  gap: 2px;
  background-color: lightgray;
}

.result {
  grid-column: 1 / 5;
  background-color: gray;
  text-align: end;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  font-size: 18px;
  color: white;
}

.result p {
  padding: 5px;
}

.zero {
  grid-column: 1 / 3;
}

.calculator button {
  cursor: pointer;
  border: none;
  font-weight: 600;
}

.calculator button:active {
  opacity: 0.8;
}

@media (min-width: 768px) {
  .calculator {
    grid-template-columns: repeat(4, 100px);
  }

  .result {
    font-size: 25px;
  }
}
