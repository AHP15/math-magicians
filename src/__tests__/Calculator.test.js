import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import buttonsInfo from '../buttons';

it('renders Calculator correctly', () => {
  const tree = renderer.create(<Calculator buttons={buttonsInfo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
