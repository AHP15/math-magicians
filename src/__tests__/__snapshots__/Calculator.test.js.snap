// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculator correctly 1`] = `
<div
  className="calculator"
>
  <div
    className="result"
    title="result"
  >
    <p>
      0
    </p>
  </div>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    AC
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    +/-
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    %
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "orange",
      }
    }
    type="button"
  >
    รท
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    7
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    8
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    9
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "orange",
      }
    }
    type="button"
  >
    x
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    4
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    5
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    6
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "orange",
      }
    }
    type="button"
  >
    -
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    1
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    2
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    3
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "orange",
      }
    }
    type="button"
  >
    +
  </button>
  <button
    className="zero"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    0
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "auto",
      }
    }
    type="button"
  >
    .
  </button>
  <button
    className=""
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "orange",
      }
    }
    type="button"
  >
    =
  </button>
</div>
`;
